src/github.com/getlantern/flashlight/client/handler.go

```diff
@@ -49,7 +49,11 @@ func (client *Client) intercept(resp http.ResponseWriter, req *http.Request) {
 		respondBadGateway(resp, fmt.Sprintf("Unable to hijack connection: %s", err))
 		return
```

Comment:
Yes I think so too. I'm prepared a commit to handle all the comments you've done in this thread.